### PR TITLE
replica set for docker

### PR DESCRIPTION
**Pull request by lmd59**
# Summary
Update docker setup to create replica set in containers. Update documentation and environment examples for local mongodb replica set.

## New behavior
Docker starts up using a replica set (and user can do local development with a replica set) allowing the use of certain db features (like transactions).

## Code changes

- README has instructions for local replica set changes.
- Environment variables are updated to reflect replica set connection. 
- docker-compose and the docker-compose example are updated to set up replica set.

# Testing guidance
- Follow the README to set up the local replica set.
- Build and restart the app.
- Verify basic application function by creating draft measures and releasing them.
- Stop your local mongodb and application and run `docker compose up --build` instead
- Verify basic application function by creating draft measures and releasing them.
- Cherry-pick the `batch draft` commit from the `batch-draft` branch (without the later replica set commit)
- Test the batch draft functionality for both local development and using docker: 

1. Temporarily change modifyResourceFields, line 39, to count < 0 for easier testing
2. Rebuild and start the application
3. Reset database and load your favorite measure bundle (I used `directory-upload.sh` to load)
4. Create a draft of the repository measure (should create a draft of the main library and the measure successfully)
5. Delete the draft measure
6. Create a draft of the repository measure again (should fail with a duplicate key error because we reduced the version increments count to 0)
7. There should still just be 1 library and 0 measures in drafts (previously this would have successfully created a new measure and only failed on the library version conflict, but now the creation of a new measure/library parent/child fails together)

### CODE REVIEW

**Review by elsaperelli — CHANGES_REQUESTED** [review]
Just some small readme changes but other than that works/looks great!

File: README.md [hunk 1]
```diff
@@ -34,6 +34,28 @@ npm install --workspace=<app-or-service> <package-name>
 
 Copy `app/.env.example` to `app/.env.local` and `service/.env.example` to `service/.env`. Make any changes to point to the measure repository service, Mongo database, and optionally the VSAC API. `0.0.0.0` may be a more appropriate database address than `localhost` for certain environment setups.
 
+### Mongo Replica Set Setup
+
+Use the mongodb configuration file to configure the single node replica set. More information about the configuration file and system location, see the mongodb [configuration file documentation](https://www.mongodb.com/docs/manual/reference/configuration-options/).
+
+1. First shutdown any currently running mongodb standalone instance: `brew services stop mongodb-community`.
```
[the remark this answers is not in the document]
Comment (lmd59) [reply README.md:41]: I think you could say either way, but why not plural? :)

[the remark this answers is not in the document]
Comment (lmd59) [reply README.md:41]:
```suggestion
1. First shutdown any currently running mongodb standalone instances: `brew services stop mongodb-community`.
```

**Review by elsaperelli — APPROVED** [review]
Lgtm!

**Pull request merged**